exporter/metric/prometheus/counter.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregator"
+)
+
+type counters struct {
+	registerer  prometheus.Registerer
+	counters    map[metricKey]prometheus.Counter
+	counterVecs map[*export.Descriptor]*prometheus.CounterVec
+}
+
+func newCounters(registerer prometheus.Registerer) counters {
```

Comment:
i like this refactor.. 

this should probably return a pointer, no? the methods on the object rightly have a pointer receiver, so might as well return a pointer and reduce a copy here.. 

same applies to histograms and gauges too..

Reply:
These are placed in structs, so the pointer methods work. I'm happy with fewer pointers, fwiw.

Reply:
We could reduce a copy using pointer or save an alloc not using a pointer. In the end both should have little to no difference since exporters are created once. 

Are you okay leaving this that way or do you have a strong objections against this?

Reply:
I actually don't know if that solves it. Copies could be done, it would just be a shallow copy, It is not like Mutex that a copy could have inconsistent and incorrect state.

Reply:
I don't see a problem, anyway.